main.tf

```diff
@@ -96,6 +89,14 @@ resource "azurerm_application_gateway" "this" {
       subnet_id                       = var.gateway_ip_configuration.subnet_id
     }
   }
+  dynamic "frontend_ip_configuration" {
+    for_each = var.create_public_ip == false && var.public_ip_resource_id == null ? [] : [1]
+
+    content {
+      name                 = coalesce(var.frontend_ip_configuration_public_name, local.frontend_ip_configuration_name)
+      public_ip_address_id = var.create_public_ip == true ? azurerm_public_ip.this[0].id : var.public_ip_resource_id
```

Comment:
I think we also need a `precondition` block on `azurerm_public_ip.this` block to ensure that either `var.create_public_ip` is `false` or `var.public_ip_resource_id` is `null`.